bundles/core/org.eclipse.smarthome.core/src/main/java/org/eclipse/smarthome/core/net/NetUtil.java

```diff
@@ -25,17 +31,64 @@
  *
  * @author Markus Rathgeb - Initial contribution and API
  * @author Mark Herwege - Added methods to find broadcast address(es)
+ * @author Stefan Triller - Converted to OSGi service with primary ipv4 conf
  */
-public class NetUtil {
+@Component(name = "org.eclipse.smarthome.network", property = { "service.config.description.uri=system:network",
+        "service.config.label=Network Settings", "service.config.category=system" })
+public class NetUtil implements NetworkAddressService {
 
+    private static final String PRIMARY_ADDRESS = "primaryAddress";
     private static final Logger LOGGER = LoggerFactory.getLogger(NetUtil.class);
 
-    private NetUtil() {
+    private static final Pattern IPV4_PATTERN = Pattern
+            .compile("^(([01]?\\d\\d?|2[0-4]\\d|25[0-5])\\.){3}([01]?\\d\\d?|2[0-4]\\d|25[0-5])$");
+
+    private String primaryAddress;
+
+    @Activate
+    protected void activate(Map<String, Object> props) {
+        modified(props);
+    }
+
+    @Modified
+    public synchronized void modified(Map<String, Object> config) {
+        String primaryAddressConf = (String) config.get(PRIMARY_ADDRESS);
+        if (primaryAddressConf == null || primaryAddressConf.isEmpty() || !isValidIPConfig(primaryAddressConf)) {
+            // if none is specified we return the default one for backward compatibility
+            LOGGER.warn("Non valid IP configuration found, will continue to use first interface");
```

Comment:
As >90% of users will have exactly one interface, there is no need to set this configuration parameter.
Why should you bother them with a warning?
You should only show a warning if the parameter is invalid.